src/sqlancer/yugabyte/ysql/YSQLProvider.java

```diff
@@ -0,0 +1,366 @@
+package sqlancer.yugabyte.ysql;
+
+import java.net.URI;
+import java.net.URISyntaxException;
+import java.sql.Connection;
+import java.sql.DriverManager;
+import java.sql.SQLException;
+import java.sql.Statement;
+import java.util.Arrays;
+
+import com.google.auto.service.AutoService;
+
+import sqlancer.AbstractAction;
+import sqlancer.DatabaseProvider;
+import sqlancer.IgnoreMeException;
+import sqlancer.MainOptions;
+import sqlancer.Randomly;
+import sqlancer.SQLConnection;
+import sqlancer.SQLProviderAdapter;
+import sqlancer.StatementExecutor;
+import sqlancer.common.DBMSCommon;
+import sqlancer.common.query.SQLQueryAdapter;
+import sqlancer.common.query.SQLQueryProvider;
+import sqlancer.common.query.SQLancerResultSet;
+import sqlancer.yugabyte.ysql.YSQLOptions.YSQLOracleFactory;
+import sqlancer.yugabyte.ysql.gen.YSQLAlterTableGenerator;
+import sqlancer.yugabyte.ysql.gen.YSQLAnalyzeGenerator;
+import sqlancer.yugabyte.ysql.gen.YSQLCommentGenerator;
+import sqlancer.yugabyte.ysql.gen.YSQLDeleteGenerator;
+import sqlancer.yugabyte.ysql.gen.YSQLDiscardGenerator;
+import sqlancer.yugabyte.ysql.gen.YSQLDropIndexGenerator;
+import sqlancer.yugabyte.ysql.gen.YSQLIndexGenerator;
+import sqlancer.yugabyte.ysql.gen.YSQLInsertGenerator;
+import sqlancer.yugabyte.ysql.gen.YSQLNotifyGenerator;
+import sqlancer.yugabyte.ysql.gen.YSQLSequenceGenerator;
+import sqlancer.yugabyte.ysql.gen.YSQLSetGenerator;
+import sqlancer.yugabyte.ysql.gen.YSQLTableGenerator;
+import sqlancer.yugabyte.ysql.gen.YSQLTableGroupGenerator;
+import sqlancer.yugabyte.ysql.gen.YSQLTransactionGenerator;
+import sqlancer.yugabyte.ysql.gen.YSQLTruncateGenerator;
+import sqlancer.yugabyte.ysql.gen.YSQLUpdateGenerator;
+import sqlancer.yugabyte.ysql.gen.YSQLVacuumGenerator;
+import sqlancer.yugabyte.ysql.gen.YSQLViewGenerator;
+
+@AutoService(DatabaseProvider.class)
+public class YSQLProvider extends SQLProviderAdapter<YSQLGlobalState, YSQLOptions> {
+
+    public static final Object CREATION_LOCK = new Object();
```

Comment:
Perhaps you could add a comment on why we need this lock?

Reply:
Added comment and changed variable name